moveit_ros/planning/planning_scene_monitor/test/current_state_monitor_tests.cpp

```diff
@@ -156,6 +157,36 @@ TEST(CurrentStateMonitorTests, NoModelTest)
                std::invalid_argument);
 }
 
+TEST(CurrentStateMonitorTests, HaveCompleteStateConstructFalse)
+{
+  // GIVEN a CurrentStateMonitor
+  planning_scene_monitor::CurrentStateMonitor current_state_monitor{
+    std::make_unique<MockMiddlewareHandle>(), moveit::core::loadTestingRobotModel("panda"),
+    std::make_shared<tf2_ros::Buffer>(std::make_shared<rclcpp::Clock>())
+  };
+
+  // WHEN it is constructed
+  // THEN we expect haveCompleteState to be false
```

Comment:
nit: this weird capitalization is weird/hard to read. Not sure it adds anything.  WHEN, THEN, etc.

Reply:
Does Doxygen support pulling out GWT conditions in some format? That would be a nice motivation for this explicit formatting.

Reply:
The given-when-then comes from BDD (https://en.wikipedia.org/wiki/Behavior-driven_development) and is a practice of testing.  I'll be the first to admit I don't fully understand BDD but that it does seem to provide helpful patterns for writing nice tests.